python/FindPythonLibs.cmake

```diff
@@ -194,33 +253,22 @@ set(PYTHON_DEBUG_LIBRARIES "${PYTHON_DEBUG_LIBRARY}")
 # what SELECT_LIBRARY_CONFIGURATIONS() expects.
 set(PYTHON_LIBRARY_DEBUG "${PYTHON_DEBUG_LIBRARY}")
 set(PYTHON_LIBRARY_RELEASE "${PYTHON_LIBRARY}")
-include(SelectLibraryConfigurations)
+include(${CMAKE_CURRENT_LIST_DIR}/SelectLibraryConfigurations.cmake)
 SELECT_LIBRARY_CONFIGURATIONS(PYTHON)
+
+if(PYTHON_LIBRARY AND NOT PYTHON_LIBRARIES)
+  set(PYTHON_LIBRARIES "${PYTHON_LIBRARY}")
+endif()
 # SELECT_LIBRARY_CONFIGURATIONS() sets ${PREFIX}_FOUND if it has a library.
 # Unset this, this prefix doesn't match the module prefix, they are different
 # for historical reasons.
 unset(PYTHON_FOUND)
 
-# This is a hack to make FindPackageHandleStandardArgs not failing
-# when EXACT version of the form "3" or "3.2" is required and
-# PYTHONLIBS_VERSION_STRING is "3.2.4".
-# This hack is in CMake 3.2 and above.
-set(_PYTHONLIBS_VERSION_STRING "${PYTHONLIBS_VERSION_STRING}")
-if(PythonLibs_FIND_VERSION)
-    set(PYTHONLIBS_VERSION_STRING "${PythonLibs_FIND_VERSION_MAJOR}")
-    if(PythonLibs_FIND_VERSION_COUNT GREATER 1)
-        set(PYTHONLIBS_VERSION_STRING "${PYTHONLIBS_VERSION_STRING}.${PythonLibs_FIND_VERSION_MINOR}")
-    endif(PythonLibs_FIND_VERSION_COUNT GREATER 1)
-    if(PythonLibs_FIND_VERSION_COUNT GREATER 2)
-        set(PYTHONLIBS_VERSION_STRING "${PYTHONLIBS_VERSION_STRING}.${PythonLibs_FIND_VERSION_TWEAK}")
-    endif(PythonLibs_FIND_VERSION_COUNT GREATER 2)
-endif()
-
+# include(${CMAKE_CURRENT_LIST_DIR}/FindPackageHandleStandardArgs.cmake)
```

Comment:
Why remove the hack and why this dead code?

Reply:
a correct implementation is provided in the directory now